modern/org/commcare/modern/database/DatabaseHelper.java

```diff
@@ -106,11 +107,18 @@ private static void addDataToValues(HashMap<String, Object> values,
 
         if(e instanceof IMetaData) {
             IMetaData m = (IMetaData)e;
-            for(String key : m.getMetaDataFields()) {
+            for (String key : m.getMetaDataFields()) {
                 Object o = m.getMetaData(key);
-                if(o == null ) { continue;}
-                String value = o.toString();
-                values.put(TableBuilder.scrubName(key), value);
+                if (o == null) {
+                    continue;
+                }
+                String scrubbedKey = TableBuilder.scrubName(key);
+                if (o instanceof Date) {
+                    // store date as seconds since epoch
+                    values.put(scrubbedKey, ((Date)o).getTime());
```

Comment:
Any metadata field that is a `Date` will now be stored as seconds from epoch. This fits nicely with SQLite's Date model, or lack of one. I've verified that the only metadata date field in `commcare-core` and `commcare-android` is in `FormRecord` (and `AndroidLogEntry`, which never reads the metadata value from the DB). @wpride do you use date metadata fields CloudCare-exclusive models?